CHANGELOG.md

```diff
@@ -2,7 +2,7 @@
 
 ## [Unreleased]
 
-- No changes yet.
+- Add `buf-cli-warning-bin` header to detect and print server warning messages.
```

Comment:
Maybe this is overthinking it but do we want this to be an error detail to be able to provide more details in the future? Perhaps we start with warning message(s) but maybe at some point it adds other metadata (reference id for support, etc.)?

Reply:
Totally fair question, and here are some thoughts:

- The end result is a message string printed to the user, so whether it's a blob in a header or an error detail shouldn't matter
- Headers are easier to work with and are universally supported (maybe length? considerations). 
- If (extremely unlikely) there is a change in error details, then future servers would need to remain backwards-compatible with older clients
- It keeps it consistent with what we're doing with raw errors in `NewErrorInterceptor`, dumping the raw errors whichever way the server decides to format it (moot point because the same could be true for a base64-encoded string field)